lib/arqo/query.rb

```diff
@@ -12,12 +12,43 @@ def initialize(relation = derived_relation)
 
     private
 
-    def derived_relation_name
-      self.class.name.sub(/Query$/, '')
+    # Returns the model to which the query object is associated.
+    # By default it will infer the name from the query object's class
+    # assuming the name is the name of the model with "Query" as the suffix.
+    #
+    # As an example, UserQuery would be associated with the User model and
+    # Blog::PostQuery would be associated with the Blog::Post namespaced model.
+    #
+    # This method can be overridden to bypass this convention and associate
+    # the query object to another relation by returning it.
+    #
+    # ==== Example
+    #     class PostQuery < ARQO::Query
+    #       # ...
+    #
+    #       private
+    #
+    #       def associated_relation
+    #         Blog::Post
+    #       end
+    #     end
+    #
+    # @return [Class, ActiveRecord::Relation]
+    def associated_relation
```

Comment:
maybe `model_name` sounds simpler? What do you think?

Reply:
nevermind, I just saw that it could be something more than just a name

Reply:
Yeah I didn’t want it to expect a string but I’m not happy with the current name

Reply:
What about `model_klass` or just `klass` ?

Reply:
That could be misleading since it can also return an `ActiveRecord::Relation`. I even think it'd be more common to use It with relations rather than comments. Like

```ruby
class CensoredPostQuery < ARQO::Query
  module Scope
  end
  
  private

  def associated_relation
    Comment.where(censored: true)
  end
end
```